presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/AddLocalExchanges.java

```diff
@@ -325,6 +325,16 @@ public PlanWithProperties visitAggregation(AggregationNode node, StreamPreferred
                 // !isPresent() indicates the property was satisfied completely
                 preGroupedSymbols = groupingKeys;
             }
+            else {
+                // Enable segmented streaming aggregation if the input is pre-grouped by GroupBy key prefix.
```

Comment:
This requirement might be too strong. It is not necessary for the input to be grouped on the prefix of the grouping keys. It is sufficient for the input to be grouped on a subset of the grouping keys.

Reply:
That is right. As long as the prefix of the sorted columns is a subset of the group by column, we can enable segmented aggregation.

The problem here is that presto table writer doesn't support group by property, so we have to create the table with sort_by columns and infer the groupby property. In this case, the rows are grouped by the prefix of the sort_by columns. We don't want to expose all prefix column combinations in the data property. Only the full sort_by column list is in the sort property.

The logic below need to be changed to iterate all possible prefixes of the sort_by columns in child.getProperties().getLocalProperties() and see if it is a subset of the groupingKeys. And we need to find the longest prefix of the sort_by columns that is a subset of the groupingKeys.